retrofit-adapters/rxjava/src/main/java/retrofit2/adapter/rxjava/Transformer.java

```diff
@@ -0,0 +1,15 @@
+package retrofit2.adapter.rxjava;
+
+import rx.Completable;
+import rx.Observable;
+
+/**
+ * A class that allows {@link Observable} and {@link Completable} to be transformed before
+ * returning then from {@link RxJavaCallAdapterFactory}.
+ *
+ * For example, you can use the {@link Transformer} to apply a observeOn or subscribeOn {@link rx.Scheduler}
+ */
+public interface Transformer {
```

Comment:
I agree, we could also go for a `TransformerProvider` which returns a Rx `Transformer` and a `CompletableTransformer` which we apply using `compose` on the `Observable` and `Completable`